d3rlpy/base.py

```diff
@@ -336,15 +340,27 @@ def fit(self,
             if new_transitions:
                 transitions = env_transitions + new_transitions
 
-            indices = np.random.permutation(np.arange(len(transitions)))
-            n_iters = len(transitions) // self.batch_size
-            range_gen = trange(n_iters) if show_progress else range(n_iters)
-            for itr in range_gen:
-                # pick transitions
-                sampled_transitions = []
-                head_index = itr * self.batch_size
-                for index in indices[head_index:head_index + self.batch_size]:
-                    sampled_transitions.append(transitions[index])
+            # shuffle data
+            if shuffle:
+                random.shuffle(transitions)
+
+            # batches generator
+            total_batches = ((len(transitions) + self.batch_size - 1) //
+                             self.batch_size)
+            batches = (transitions[i * self.batch_size:(i + 1) *
+                                   self.batch_size]
+                       for i in range(total_batches))
+
+            # Epoch progress bar
+            tqdm_epoch = tqdm(batches,
+                              total=total_batches,
+                              disable=not show_progress,
+                              desc=f'Epoch {epoch + 1}')
+
+            # dict to add incremental mean losses to epoch
+            step_incremental_losses = {}
```

Comment:
In my opinion, this name looks long, which breaks the following line. I would suggest simply `loss_history`.

Reply:
Or you can add a method to `D3RLPyLogger`, which is like `get_metrics` that returns the current buffer. Then, you can remove `step_incremental_losses`.

Reply:
ok